File: worker/src/pipeline/upload-to-storage.ts
import { getSupabase } from "../db.js";

/**
 * Uploads a markdown transcript file to Supabase Storage and returns its public URL.
 * Path follows the convention from docs/data-storage.md:
 * {shard}/{videoId}/{lang}.md where shard = first 2 chars of videoId.
 */
export async function uploadToStorage(
  videoId: string,
  language: string,
  markdownContent: string,
): Promise<string> {
  const supabase = getSupabase();
  const shard = videoId.slice(0, 2);
  const path = `${shard}/${videoId}/${language}.md`;

  const { error } = await supabase.storage
    .from("transcripts")
    .upload(path, Buffer.from(markdownContent, "utf-8"), {
      contentType: "text/markdown",
      upsert: true,
    });

  if (error) {
    throw new Error(`Storage upload failed: ${error.message}`);
  }

  const {
    data: { publicUrl },
  } = supabase.storage.from("transcripts").getPublicUrl(path);

  return publicUrl;
}

File: worker/src/pipeline/upload_to_storage.py
from __future__ import annotations

from src.db import get_supabase


def upload_to_storage(video_id: str, language: str, markdown_content: str) -> str:
    """
    Upload a Markdown transcript file to Supabase Storage and return its public URL.
    Path convention: {shard}/{videoId}/{lang}.md where shard = first 2 chars of videoId.
    """
    sb = get_supabase()
    shard = video_id[:2]
    path = f"{shard}/{video_id}/{language}.md"

    sb.storage.from_("transcripts").upload(
        path,
        markdown_content.encode("utf-8"),
        file_options={"content-type": "text/markdown", "upsert": "true"},
    )

    res = sb.storage.from_("transcripts").get_public_url(path)
    return res
